assets/components/contributionSelection/contributionSelectionReducer.js

```diff
@@ -154,7 +152,11 @@ function contributionSelectionReducerFor(scope: string, countryGroupId: CountryG
         };
 
       case 'SET_AMOUNT':
-        return updatePredefinedAmount(state, action.amount);
+        return updatePredefinedAmount(state, state.contributionType, action.amount);
+
+      case 'SET_AMOUNT_FOR_CONTRIBUTION_TYPE':
+        return updatePredefinedAmount(state, action.contributionType, action.amount);
```

Comment:
... the next case

Reply:
The difference is whether you are updating the selected amount of the currently selected contribution type (`state.contributionType`) or the selected amount of a contribution type of your choosing (`action.contributionType`).